### DISCUSSION TITLE
How is the binding done between SEGA 315-5250 and the Z80/M68000 CPUs?

### DISCUSSION THREAD

**Discussion by jotego** [mamedev/discussions#51] [Q&A] [answered]
The code suggests that device [SEGA 315-5250](https://github.com/mamedev/mame/blob/afd529ba9a3a7fe97ffb9754e91c4dbc830aa6a1/src/mame/sega/segaic16_m.h#L4) is able to interrupt the Z80 and M68000 CPUs in system 16B because of the members m_68kint_callback and m_zint_callback. But for the life of me that I cannot see where the binding is done and to which interrupt line. Nor can I see how the Z80 can access that same device to read data after its interrupted. It does not appear in the memory map in the [S16B driver](https://github.com/mamedev/mame/blob/afd529ba9a3a7fe97ffb9754e91c4dbc830aa6a1/src/mame/sega/segas16b.cpp) either.

Is this connected at all?

**Comment by cuavas** [comment, marked as answer]
Those outputs are only used on the Sega X Board, and only for one of the two timers.  It’s connected where you’d expect: https://github.com/mamedev/mame/blob/mame0256/src/mame/sega/segaxbd.cpp#L1683

**Reply by jotego** [reply]
Thank so much. That makes sense now.
